diff --git a/upstream/ffx-1.1.4/sdk/include/FidelityFX/host/ffx_types.h b/upstream/ffx-1.1.4/sdk/include/FidelityFX/host/ffx_types.h
index f7625c2..9564f3b 100644
--- a/upstream/ffx-1.1.4/sdk/include/FidelityFX/host/ffx_types.h
+++ b/upstream/ffx-1.1.4/sdk/include/FidelityFX/host/ffx_types.h
@@ -23,6 +23,12 @@
 #pragma once
 
 #include <stdint.h>
+#include <string.h>
+#include <wchar.h>
+#ifdef __cplusplus
+#include <cstring>
+#include <cwchar>
+#endif
 
 ///
 /// @defgroup ffxSDK SDK
diff --git a/upstream/ffx-1.1.4/sdk/src/backends/vk/ffx_vk.cpp b/upstream/ffx-1.1.4/sdk/src/backends/vk/ffx_vk.cpp
index b2439b6..71d5369 100644
--- a/upstream/ffx-1.1.4/sdk/src/backends/vk/ffx_vk.cpp
+++ b/upstream/ffx-1.1.4/sdk/src/backends/vk/ffx_vk.cpp
@@ -34,6 +34,8 @@
 #endif  // _WIN32
 
 #include <vulkan/vulkan.h>
+#include <new>
+#include <cmath>
 
 // prototypes for functions in the interface
 FfxVersionNumber       GetSDKVersionVK(FfxInterface* backendInterface);
diff --git a/upstream/ffx-1.1.4/sdk/src/shared/ffx_message.cpp b/upstream/ffx-1.1.4/sdk/src/shared/ffx_message.cpp
index e325a05..a31da46 100644
--- a/upstream/ffx-1.1.4/sdk/src/shared/ffx_message.cpp
+++ b/upstream/ffx-1.1.4/sdk/src/shared/ffx_message.cpp
@@ -28,6 +28,8 @@
 #define WIN32_LEAN_AND_MEAN
 #endif
 #include <windows.h>  // required for OutputDebugString()
+#include <wchar.h>
+#include <stdio.h>
 #endif                // #ifdef _WIN32
 
 static ffxMessageCallback s_messageCallback;
@@ -48,10 +50,10 @@ void ffxPrintMessage(uint32_t type, const wchar_t* message)
         // Format the message string
         wchar_t buffer[512];
         if (type == FFX_MESSAGE_TYPE_ERROR) {
-            swprintf_s(buffer, 512, L"FSR_API_DEBUG_ERROR: %ls\n", message);
+            swprintf(buffer, 512, L"FSR_API_DEBUG_ERROR: %ls\n", message);
         }
         else if (type == FFX_MESSAGE_TYPE_WARNING) {
-            swprintf_s(buffer, 512, L"FSR_API_DEBUG_WARNING: %ls\n", message);
+            swprintf(buffer, 512, L"FSR_API_DEBUG_WARNING: %ls\n", message);
         }
         OutputDebugStringW(buffer);
     } else {
